### PR TITLE
Feature slicer. 

### PR DESCRIPTION
**Pull request by msultan**
- [x] Implement feature / fix bug
 - [x] Add tests
 - [x] Update changelog

I wanted a featurizer that would only return a certain subset of feature indices. And i wanted the dataframe to work with it.

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -11,10 +11,12 @@ API Changes
 
 New Features
 ~~~~~~~~~~~~
+- Added new featurizer ```FeatureSlicer``` capable of slicing features.
```
Comment (cxhernandez) [inline docs/changelog.rst]: Wow. Just wow.

File: msmbuilder/feature_selection/featureselector.py [hunk 2]
```diff
@@ -93,3 +93,62 @@ def describe_features(self, traj):
...
         return all_res
+
+
+class FeatureSlicer(Featurizer):
+    """Extracts features (e.g. subsets) from data along the feature dimension.
+
+    Parameters
+    ----------
+    feat : MSMBuilder Featurizer object,required
+        an initaed MSMBuilder Featurizer object
```
Comment (cxhernandez) [inline msmbuilder/feature_selection/featureselector.py]: > MSMBuilder Featurizer object, required an initiated MSMBuilder Featurizer object

File: msmbuilder/feature_selection/featureselector.py [hunk 3]
```diff
@@ -93,3 +93,62 @@ def describe_features(self, traj):
...
+    -----
+    You must give either index or first (but not both)
+
+    """
+
+    def __init__(self, feat=None, indices=None):
+
+        if feat is None:
+            raise ValueError("Please provide a fitted"
+                             "featurizer object")
```
Comment (cxhernandez) [inline msmbuilder/feature_selection/featureselector.py]: add a space after fitted

File: msmbuilder/feature_selection/featureselector.py [hunk 4]
```diff
@@ -93,3 +93,62 @@ def describe_features(self, traj):
...
+
+        if indices is None:
```
Comment (cxhernandez) [inline msmbuilder/feature_selection/featureselector.py]: Might also want to check if indices is array_like

**Comment by brookehus** [comment]
what does it mean to supply either index or first? What is first?

**Comment by msultan** [comment]
woops, thanks for the catch @brookehus. made some more fixes to the docs.

**Comment by brookehus** [comment]
so if you give one int you just give a feature? would like to see a test for every type of slice input (list, array, integer)

**Comment by msultan** [comment]
@brookehus that doesnt matter since i am casting all of them to array internally so an int gets mapped to an array of size 1.

File: msmbuilder/feature_selection/featureselector.py [hunk 5]
```diff
@@ -93,3 +93,62 @@ def describe_features(self, traj):
...
+            raise ValueError("Please specify indices")
+        if not (isinstance(indices, list)
+                or isinstance(indices, np.ndarray)
+                or isinstance(indices, np.int)):
+            raise ValueError("Type of indices is neither a list/array "
+                             "nor an int.")
+        self.feat = feat
+        self.indices = np.array(indices)
```
Comment (msultan) [inline msmbuilder/feature_selection/featureselector.py]: This is what i mean with the array mapping. 
``` python
a=np.arange(5)
assert a[np.array(2)] == 2

```

**Pull request merged**